src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -410,10 +410,12 @@ private trait DemuxWriter extends Closeable {
 /** A writer that writes [[DemuxRecord]]s as [[SamRecord]]s. */
 private class SamRecordWriter(output: PathToBam,
                               val header: SAMFileHeader,
-                              val umiTag: String) extends DemuxWriter {
+                              val umiTag: String,
+                              val numSamples: Int) extends DemuxWriter {
   val order = if (header.getSortOrder == SortOrder.unsorted) None else SamOrder(header)
   private val writer = SamWriter(output, header, sort=order,
-    async = DemuxFastqs.UseAsyncIo, maxRecordsInRam = DemuxFastqs.MaxRecordsInRamPerSamFileWriter)
+    async = DemuxFastqs.UseAsyncIo,
+    maxRecordsInRam = Math.max(10000,  DemuxFastqs.MaxRecordsInRamPerSamFileWriter / numSamples))
```

Comment:
I think this constant should get renamed to just `MaxRecordsInRam` or `MaxRecordsInRamWhenSortingSams`, since it's now treated as a total and divided by the number of samples.